doc/source/conf.py

```diff
@@ -435,6 +435,11 @@ def linkcode_resolve(domain, info):
         if not fn:
             return None
 
+        # Ignore re-exports as their source files are not within the numpy repo
+        module = inspect.getmodule(obj)
+        if module is not None and not module.__name__.startswith("numpy"):
+            return None
+
```

Comment:
What does this improve, is it needed for ctypes?

Reply:
https://github.com/numpy/numpy/blob/a525954b7e1f6bb8237e7d491a9c1d50837eeac7/doc/source/conf.py#L459-L460

Previously it would always attempt to generate a link to the respective source code within the numpy repo.
This will however generate an invalid URL if the source is located in a different package, as can be seen in 
the source of [`np.errstate.__call__`](https://numpy.org/doc/stable/reference/generated/numpy.errstate.__call__.html#numpy.errstate.__call__) for example.

Reply:
nice catch